Error: The terraform-provider-azurerm_v3.43.0_x5 plugin crashed!

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.3.8

### AzureRM Provider Version

3.43.0

### Affected Resource(s)/Data Source(s)

azurerm_role_definition

### Terraform Configuration Files

```hcl
resource "azurerm_role_definition" "roles" {
  for_each = local.role_map
  name        = each.value.name
  scope       = data.azurerm_subscription.primary.id

  permissions {
    actions     = each.value.actions
    not_actions = each.value.not_actions
    data_actions = each.value.data_actions
    not_data_actions =  each.value.not_data_actions
  }

  assignable_scopes = each.value.assignable_scopes
}
```


### Debug Output/Panic Output

```shell
Stack trace from the terraform-provider-azurerm_v3.43.0_x5 plugin:
panic: interface conversion: interface {} is nil, not string
goroutine 115 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/authorization.expandRoleDefinitionAssignableScopes(0xc001974c10?)
	github.com/hashicorp/terraform-provider-azurerm/internal/services/authorization/role_definition_resource.go:469 +0x226
github.com/hashicorp/terraform-provider-azurerm/internal/services/authorization.resourceArmRoleDefinitionCreate(0xc000ff5c80, {0x617aa80?, 0xc002458000})
	github.com/hashicorp/terraform-provider-azurerm/internal/services/authorization/role_definition_resource.go:148 +0x349
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x74a9540?, {0x74a9540?, 0xc0016f8360?}, 0xd?, {0x617aa80?, 0xc002458000?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:695 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0005e1a40, {0x74a9540, 0xc0016f8360}, 0xc0023f08f0, 0xc000ff5b00, {0x617aa80, 0xc002458000})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:837 +0xa85
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0009e1908, {0x74a9540?, 0xc0016f8240?}, 0xc000db81e0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/grpc_provider.go:1021 +0xe8d
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc000731860, {0x74a9540?, 0xc0004c3950?}, 0xc002796f50)
	github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/tf5server/server.go:818 +0x574
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x68a5140?, 0xc000731860}, {0x74a9540, 0xc0004c3950}, 0xc002796ee0, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000016000, {0x74b98c0, 0xc0016d04e0}, 0xc000a4bd40, 0xc00169f8c0, 0xb98f020, 0x0)
	google.golang.org/grpc@v1.51.0/server.go:1340 +0xd23
google.golang.org/grpc.(*Server).handleStream(0xc000016000, {0x74b98c0, 0xc0016d04e0}, 0xc000a4bd40, 0x0)
	google.golang.org/grpc@v1.51.0/server.go:1713 +0xa2f
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.51.0/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.51.0/server.go:963 +0x28a
Error: The terraform-provider-azurerm_v3.43.0_x5 plugin crashed!
```


### Expected Behaviour

azurerm_role_definition resources created successfully

### Actual Behaviour

Error: Plugin did not respond

### Steps to Reproduce

terraform init, terraform plan, terraform apply

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/authorization/role_definition_resource.go
+++ b/internal/services/authorization/role_definition_resource.go
@@ -466,7 +466,9 @@ func expandRoleDefinitionAssignableScopes(d *pluginsdk.ResourceData) []string {
 		scopes = append(scopes, assignedScope)
 	} else {
 		for _, scope := range assignableScopes {
-			scopes = append(scopes, scope.(string))
+			if s, ok := scope.(string); ok {
+				scopes = append(scopes, s)
+			}
 		}
 	}
 

